share/patch/patch_db_zonemaster_backend_ver_11.0.3.pl

```diff
@@ -96,19 +96,24 @@ sub _update_data_nomalize_domains {
     my $progress = 0;
 
     while ( my $row = $sth1->fetchrow_hashref ) {
-        my $hash_id = $row->{hash_id};
-        my $raw_params = decode_json($row->{params});
-        my $domain = $raw_params->{domain};
+        eval {
+            my $hash_id = $row->{hash_id};
+            my $raw_params = decode_json($row->{params});
+            my $domain = $raw_params->{domain};
 
-        # This has never been cleaned
-        delete $raw_params->{user_ip};
+            # This has never been cleaned
+            delete $raw_params->{user_ip};
 
-        my $params = $db->encode_params( $raw_params );
-        my $fingerprint = $db->generate_fingerprint( $raw_params );
+            my $params = $db->encode_params( $raw_params );
+            my $fingerprint = $db->generate_fingerprint( $raw_params );
 
-        $domain = Zonemaster::Backend::DB::_normalize_domain( $domain );
+            $domain = Zonemaster::Backend::DB::_normalize_domain( $domain );
 
-        $db->dbh->do('UPDATE test_results SET domain = ?, params = ?, fingerprint = ? where hash_id = ?', undef, $domain, $params, $fingerprint, $hash_id);
+            $db->dbh->do('UPDATE test_results SET domain = ?, params = ?, fingerprint = ? where hash_id = ?', undef, $domain, $params, $fingerprint, $hash_id);
+        };
+        if ($@) {
+            warn "Caught error while updating record, ignoring: $@\n";
```

Comment:
Would it be helpful to explicitly include the hash id in this message?

Reply:
I updated it to include the hash id,
```
Configured database engine: SQLite
Starting database migration

-> (1/2) Populating new result_entries table
Will update 0 rows
Progress update: 0 / 0

-> (2/2) Normalizing domain names
Will update 5 rows
Caught error while updating record with hash id b212fc97d29d51f3, ignoring: Caught Zonemaster::Backend::Error::Internal in the `Zonemaster::Backend::DB::_normalize_domain` method: Normalizing domain returned errors. Context: ['Le nom de domaine contient plusieurs points successifs.']
20%
40%
60%
80%
100%

Migration done
```